amacc.c

```diff
@@ -437,6 +439,30 @@ void expr(int lev)
             else fatal("bad lvalue in assignment");
             expr(Assign); *++e = ((ty = t) == CHAR) ? SC : SI;
             break;
+        case AddAssign: // right assoc
```

Comment:
Since there are common lines in the following, is it possible to avoid duplicated implementation?

Reply:
We can follow the order of `AddAssign`, `SubAssign`, and `MulAssign` as how `Add`, `Sub`, and `Mul` enumerate. Therefore, we can rewrite as following:

```C
case AddAssign: case SubAssign: case MulAssign:
    next();
    lastcmd = *e;
    if (*e == LC || *e == LI) {
        *e = PSH; *++e = lastcmd; *++e = PSH;
    } else fatal("bad lvalue in sub assignment");
    expr(tk); *++e = ADD + (tk - AddAssign); ty = INT; *++e = SI;
    break;
```